### DISCUSSION TITLE
Throughput in addition to RPS

### DISCUSSION THREAD

**Discussion by michaelplavnik** [VIP-Logic/NBomber#730] [Ideas]
In older enterprise integration scenarios, request might contains a batch/bulk of transaction items. As a result of the operation items can be marked with completion status. 

The idea is to support a separate stat for Throughput. Step can report on success, duplication, or error of transactions as counters. Scenario stats will then contain both total of counters and throughput per status.

**Comment by AntyaDev** [comment]
I guess maybe this dimension of Throughput could be modeled as custom metric? For example, inside the step execution, you could just use some interface that allows you to record additional metrics: success | duplication
Does it make sense?

**Comment by michaelplavnik** [comment]
Do you have example of a custom metric ?

**Comment by AntyaDev** [comment]
no, but It could look like
```csharp
var step1 = await Step.Run("login", context, async () =>
{
    context.Metric.Counter.Increment("success");

    context.Metric.Counter.Increment("success", 10);

    context.Metric.Counter.Decrement("success");
    
    return Response.Ok();
});
```

**Reply by michaelplavnik** [reply]
Custom metric would be great. Do you have plans to release it ?

**Reply by AntyaDev** [reply]
Yes, we do plan them.
We even partially implemented them, but they still need to be completed. 
**ScenarioStats** could provide this data for assertions.

**Reply by AntyaDev** [reply]
@michaelplavnik 
https://github.com/PragmaticFlow/NBomber/issues/732
